hail/src/main/scala/is/hail/annotations/Region.scala

```diff
@@ -269,16 +265,17 @@ object Region {
     case t: PBaseStruct => (addr, v) => Region.copyFrom(coerce[Long](v), addr, t.byteSize)
   }
 
-  def stagedCreate(blockSize: Size): Code[Region] =
-    Code.invokeScalaObject2[Int, RegionPool, Region](Region.getClass, "apply", asm4s.const(blockSize), Code._null)
+  def stagedCreate(blockSize: Size, pool: Code[RegionPool]): Code[Region] =
+    Code.invokeScalaObject2[Int, RegionPool, Region](Region.getClass, "apply", asm4s.const(blockSize), pool)
 
   def apply(blockSize: Region.Size = Region.REGULAR, pool: RegionPool = null): Region = {
-    (if (pool == null) RegionPool.get else pool)
-      .getRegion(blockSize)
-  }
-
-  def makeNamed(blockSize: Region.Size = Region.REGULAR, pool: RegionPool = null): Region = {
-    (if (pool == null) RegionPool.get else pool)
+    (if (pool == null) {
+      val htc = HailTaskContext.get()
+      if (htc == null) {
+        throw new IllegalStateException(s"RegionPool requested but HailTaskContext was null. On worker = ${TaskContext.get() != null}")
+      }
+      htc.getRegionPool()
+    } else pool)
```

Comment:
We maybe just shouldn't have a default `Region` with null pool constructor anymore. Seems better to just explicitly make someone grab a pool. Otherwise too easy to do the wrong thing.

Reply:
I agree.